### DISCUSSION TITLE
Feature request: support negation operator for .quartoignore

### DISCUSSION THREAD

**Discussion by jimjam-slam** [quarto-dev/quarto-cli#4054] [Q&A] [answered]
Starter templates can [include a .quartoignore](https://quarto.org/docs/extensions/starter-templates.html#creating-a-template) file to manually specify which files to exclude from the template. It says it uses a `.gitignore`-type syntax, but one omission from this is that it doesn't appear to support a negation operator `!` to force Quarto to include a file that would otherwise be excluded using its default settings.

In my case, I'd like to include a `.gitignore` with my template, as it's designed to work with other file types, and I'd like to suggest to my users which files they should ignore (eg. `node_modules`) and which they should version control (eg. `.luarc.json`).

It'd be great if I could add `!.gitignore` to `.quartoignore` to force my suggested git ignore to be included with the template!

**Comment by dragonstyle** [comment, marked as answer]
TY for the detailed write outline, I agree that this does seem like an oversight. I've opened an issue and will try to get to it soon!

**Reply by jimjam-slam** [reply]
Thanks very much! 😊

**Comment by mcanouil** [comment]
Closing this as completed and in favour of https://github.com/quarto-dev/quarto-cli/issues/4061
